tools/robotick-launcher/src/robotick/launcher/actions/launch/sync_dependencies.py

```
@@ -36,6 +47,181 @@ def _resolve_dependencies_dir(config, platform: str) -> Path:
+    asset = getattr(src, "asset", None)
+    pin = getattr(src, "pin", None)
+    if not (repo and asset and pin):
+        raise ValueError(
+            "git_source_archive source requires either 'url' or repo/pin/asset fields."
+        )
+    base = _normalize_repo_url(repo)
+    return f"{base}/releases/download/{pin}/{asset}"
+
+
+def _download_url_to_path(url: str, dest: Path) -> None:
+    with urlopen(url) as response, dest.open("wb") as fh:
```

Ruff (0.14.8)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)